setup

```diff
@@ -185,7 +200,7 @@ superbbootstrap_install_packages() {
 		eval "install_command=\"\$${packagemanager}_COMMAND\""
 
 		# Find packages with mark
-		packages_installing=$(printf '%s' "$packages_list" | sed -e "s/^.*$mark://g" -e 's/[[:space:]].*$//g' -e '/^$/d' | tr '\n' ' ')
+		packages_installing=$(printf '%s' "$packages_list" | sed -rn "s/$mark:([0-9a-zA-Z-]+)(.*)/\1/p"  | tr '\n' ' ')
```

Comment:
@weiss-d Nice, but `sed -r` is not `POSIX`-compatible thus make the script not portable to run on `BSD` machine.

```
SED(1P)               POSIX Programmer's Manual               SED(1P)

PROLOG
       This  manual  page  is  part of the POSIX Programmer's Manual.
       The Linux implementation of this interface may differ (consult
       the  corresponding  Linux manual page for details of Linux be‐
       havior), or the interface may not be implemented on Linux.

NAME
       sed — stream editor

SYNOPSIS
       sed [-n] script [file...]

       sed [-n] -e script [-e script]... [-f script_file]... [file...]

       sed [-n] [-e script]... -f script_file [-f script_file]... [file...]
```

###### No `-r` flag found...

Reply:
Well, I see.
But does the initial command work correctly on BSDs? If yes, we could just make an `if` switch here.

Reply:
You could do this by chaining a `d` sed command in there. I'll try to take a look, see if I can optimize all this.

Reply:
I ended up with this :
```bash
packages_installing=$(printf '%s' "$packages_list" | sed -e "/^.*$mark:.*\$/!d;s/.*$mark://g" -e 's/[[:space:]].*$//g;/^$/d' | tr '\n' ' ')

```

and you can tweak slightly the way comments are stripped with `packages_list=$(sed -e 's/#.*$//g;/^$/d' "$1")`, this strips empty lines directly so they don't have to be processed in each iteration.

Reply:
@LoricAndre big thanks! Added your variant in  7b18550. Tested in Arch.